### ISSUE TITLE
Bootstrap Agent: Responsibility for Launching Switch zone needs to happen earlier

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#1960]
See 11/18 Control Plane Sync for details.

TL:DR:
- Current launching of the "Switch Zone" is managed by the sled agent (nit: NOT the bootstrap agent). The sled agent monitors devinfo for the `/dev/tofino` device, and creates the switch zone with a corresponding underlay address for the switch zone to use.
- **This is wrong**. The switch zone needs to launch before trust quorum has been established, and before the sled agent is up-and-running. As such, this hardware monitoring and device launching needs to happen within the bootstrap agent itself.
- Furthermore, we will want to supply some mechanism for the underlay address to be "added later" to the switch zone so Nexus can communicate with switch zone services (i.e., MGS, Dendrite) after underlay addressing has been  established. This is tracked by https://github.com/oxidecomputer/omicron/issues/1961
